drivers/awinic/aw20216s.c

```diff
@@ -0,0 +1,136 @@
+/* Copyright 2017 Jason Williams
+ * Copyright 2018 Jack Humbert
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "quantum.h"
+#include "spi_master.h"
+#include "aw20216s.h"
+#include <ch.h>
+#include <hal.h>
+
+#define COMMAND_CHIP_ID 0xA0
+#define US_POR_DELAY 5000    // Min is 2000 us
+#define US_ENABLE_DELAY 500  // Min is  100 us
+#define PAGE_0 0
+#define PAGE_1 1
+#define PAGE_2 2
+#define PAGE_3 3
+#define PAGE_0_REG_GCR 0x00
+#define REG_GCR_BIT_CHIPEN_MASK 0x01
+#define PAGE_0_REG_GCCR 0x01
+#define US_BETWEEN_REG_WRITE_DELAY 100
+#define PWM_REGISTER_COUNT 216
+
+#define SPI_CLOCK_SPEED 2000000
+#define SPI_MODE 3
+//#define SPI_DIVISOR               (F_CPU / SPI_CLOCK_SPEED)
+#define SPI_DIVISOR 4  // around 8 MHz (smooth transitions)
```

Comment:
I'm not sure if the settings above need to be set to those specific values.  If so, then they're fine.   Otherwise, the ones that okay to be configured should be set like these: 


```suggestion
#ifndef SPI_CLOCK_SPEED
#    define SPI_CLOCK_SPEED 2000000
#endif
#ifndef SPI_MODE
#    define SPI_MODE 3
#ifndef SPI_DIVISOR
#    define SPI_DIVISOR 4 
#endif
```

Reply:
None of these things are necessary other than `SPI_DIVISOR`.
`SPI_MODE 3` is odd, it's pretty rare to see devices that don't just use mode 0, and the existing implementation works fine with mode 0.
`SPI_CLOCK_SPEED` isn't used anywhere and shouldn't be used at all because that's not how `spi_start()` expects the speed to be defined anyways.

Reply:
> 
> 
> None of these things are necessary other than `SPI_DIVISOR`.
> `SPI_MODE 3` is odd, it's pretty rare to see devices that don't just use mode 0, and the existing implementation works fine with mode 0.
> `SPI_CLOCK_SPEED` isn't used anywhere and shouldn't be used at all because that's not how `spi_start()` expects the speed to be defined anyways.

Hi @Gigahawk, you are right, this should be mode 0. Thanks.